NorthstarDLL/mods/modsavefiles.cpp

```diff
@@ -0,0 +1,515 @@
+				"save folder.",
+				fileName,
+				mod->Name)
+				.c_str());
+		return SQRESULT_ERROR;
+	}
+
+	g_pSaveFileManager->DeleteFileAsync<context>(dir / fileName);
+	return SQRESULT_NOTNULL;
+}
+
+// The param is not optional because that causes issues :)
+ADD_SQFUNC("array<string>", NS_InternalGetAllFiles, "string path", "", ScriptContext::CLIENT | ScriptContext::UI | ScriptContext::SERVER)
+{
+	// depth 1 because this should always get called from Northstar.Custom
+	Mod* mod = g_pSquirrel<context>->getcallingmod(sqvm, 1);
+	fs::path dir = savePath / fs::path(mod->m_ModDirectory).filename();
+	std::string pathStr = g_pSquirrel<context>->getstring(sqvm, 1);
+	fs::path path = dir;
+	if (pathStr != "")
+		path = dir / pathStr;
+	if (!IsPathSafe(pathStr, dir))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm,
+			fmt::format(
+				"File name invalid ({})! Make sure it does not contain any non-ASCII character, and results in a path inside your mod's "
+				"save folder.",
+				pathStr,
+				mod->Name)
+				.c_str());
+		return SQRESULT_ERROR;
+	}
+	try
+	{
+		g_pSquirrel<context>->newarray(sqvm, 0);
+		for (const auto& entry : fs::directory_iterator(path))
+		{
+			g_pSquirrel<context>->pushstring(sqvm, entry.path().filename().string().c_str());
+			g_pSquirrel<context>->arrayappend(sqvm, -2);
+		}
+		return SQRESULT_NOTNULL;
+	}
+	catch (std::exception ex)
+	{
+		spdlog::error("DIR ITERATE FAILED! Is the path valid?");
+		g_pSquirrel<context>->raiseerror(sqvm, ex.what());
+		return SQRESULT_ERROR;
+	}
+}
+
+ADD_SQFUNC("bool", NSIsFolder, "string path", "", ScriptContext::CLIENT | ScriptContext::UI | ScriptContext::SERVER)
+{
+	Mod* mod = g_pSquirrel<context>->getcallingmod(sqvm);
+	fs::path dir = savePath / fs::path(mod->m_ModDirectory).filename();
+	std::string pathStr = g_pSquirrel<context>->getstring(sqvm, 1);
+	fs::path path = dir;
+	if (pathStr != "")
+		path = dir / pathStr;
+	if (!IsPathSafe(pathStr, dir))
+	{
+		g_pSquirrel<context>->raiseerror(
+			sqvm,
+			fmt::format(
+				"File name invalid ({})! Make sure it does not contain any non-ASCII character, and results in a path inside your mod's "
+				"save folder.",
+				pathStr,
+				mod->Name)
+				.c_str());
+		return SQRESULT_ERROR;
+	}
+	try
+	{
+		g_pSquirrel<context>->pushbool(sqvm, fs::is_directory(path));
+		return SQRESULT_NOTNULL;
+	}
+	catch (std::exception ex)
+	{
+		spdlog::error("DIR READ FAILED! Is the path valid?");
+		spdlog::info(path.string());
+		g_pSquirrel<context>->raiseerror(sqvm, ex.what());
+		return SQRESULT_ERROR;
+	}
+}
+
+// ok, I'm just gonna explain what the fuck is going on here because this
+// is the pinnacle of my stupidity and I do not want to touch this ever
+// again & someone will eventually have to maintain this.
+
+// P.S. if you don't want me to do the entire table -> JSON -> string thing...
+// Fix it in scriptjson first k thx bye
```

Comment:
Wasn't that fix merged already?

Reply:
that fix was never made. it's a completely different thing.